From e5def7b2a1709b3ee7f91df6f9f34818e14523c6 Mon Sep 17 00:00:00 2001
From: Omri Itach <omrii@marvell.com>
Date: Tue, 28 Nov 2017 11:39:24 +0200
Subject: [PATCH 2090/2241] dts: cp110: remove duplicated files for multiple
 cp110 instances

Armada-7k/8k included armada-cp110.dtsi file twice, using specific file
per each CP110: armada-cp110-0.dtsi & armada-cp110-1.dtsi.

Armada-8kPlus family supports up to 8 CPs, so using above method would
have required duplicating these files for each CP in the system.

To avoid this duplication, this patch removes the duplicated files by
slightly modifying armada-cp110.dtsi, and convert all properties that
differ amongst CPs, to be set with new definitions, and defining those
definitions in the files including them (before the include), and
un-defining them after the include.

Change-Id: I7d4458674efb93f12126b861d6fee347ba4e3213
Signed-off-by: Omri Itach <omrii@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/46954
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Tested-by: Hanna Hawa <hannah@marvell.com>
---
 .../arm64/boot/dts/marvell/armada-3900-db-axis.dts |    2 +-
 arch/arm64/boot/dts/marvell/armada-3900-db.dtsi    |    2 +-
 arch/arm64/boot/dts/marvell/armada-7020-amc.dts    |    2 +-
 arch/arm64/boot/dts/marvell/armada-7020.dtsi       |   46 +-
 arch/arm64/boot/dts/marvell/armada-7040-db-B.dts   |    2 +-
 arch/arm64/boot/dts/marvell/armada-7040-db-C.dts   |    2 +-
 arch/arm64/boot/dts/marvell/armada-7040-db-D.dts   |    2 +-
 arch/arm64/boot/dts/marvell/armada-7040-db-E.dts   |    2 +-
 .../boot/dts/marvell/armada-7040-db-all-pcie.dts   |    2 +-
 .../arm64/boot/dts/marvell/armada-7040-db-nand.dts |    2 +-
 .../dts/marvell/armada-7040-db-pp2-vanilla.dts     |    2 +-
 arch/arm64/boot/dts/marvell/armada-7040-db.dts     |    2 +-
 arch/arm64/boot/dts/marvell/armada-7040-db.dtsi    |    2 +-
 arch/arm64/boot/dts/marvell/armada-7040-pcac.dts   |    2 +-
 arch/arm64/boot/dts/marvell/armada-7040.dtsi       |   55 +-
 .../boot/dts/marvell/armada-70x0-customer.dts      |    2 +-
 arch/arm64/boot/dts/marvell/armada-8020.dtsi       |   72 +-
 arch/arm64/boot/dts/marvell/armada-8040-db-B.dts   |    4 +-
 arch/arm64/boot/dts/marvell/armada-8040-db-C.dts   |    4 +-
 arch/arm64/boot/dts/marvell/armada-8040-db-D.dts   |    4 +-
 arch/arm64/boot/dts/marvell/armada-8040-db-E.dts   |    4 +-
 .../dts/marvell/armada-8040-db-pp2-vanilla.dts     |    4 +-
 arch/arm64/boot/dts/marvell/armada-8040-db.dts     |    4 +-
 arch/arm64/boot/dts/marvell/armada-8040-db.dtsi    |    4 +-
 .../dts/marvell/armada-8040-mcbin-pp2-vanilla.dts  |    4 +-
 arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts  |    4 +-
 arch/arm64/boot/dts/marvell/armada-8040-ocp.dts    |    4 +-
 arch/arm64/boot/dts/marvell/armada-8040.dtsi       |   76 +-
 .../boot/dts/marvell/armada-80x0-customer.dts      |    4 +-
 arch/arm64/boot/dts/marvell/armada-ap806.dtsi      |    9 +
 arch/arm64/boot/dts/marvell/armada-cp110-0.dtsi    |   68 --
 arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi    |   59 -
 arch/arm64/boot/dts/marvell/armada-cp110.dtsi      | 1156 ++++++++++----------
 33 files changed, 870 insertions(+), 743 deletions(-)
 delete mode 100644 arch/arm64/boot/dts/marvell/armada-cp110-0.dtsi
 delete mode 100644 arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi

diff --git a/arch/arm64/boot/dts/marvell/armada-3900-db-axis.dts b/arch/arm64/boot/dts/marvell/armada-3900-db-axis.dts
index fe92e13..4f8611a 100644
--- a/arch/arm64/boot/dts/marvell/armada-3900-db-axis.dts
+++ b/arch/arm64/boot/dts/marvell/armada-3900-db-axis.dts
@@ -47,7 +47,7 @@
 	compatible = "marvell,armada3900-db-axis", "marvell,armada7040",
 		     "marvell,armada-ap806-quad", "marvell,armada-ap806";
 
-	cp110-0 {
+	cp0 {
 		config-space {
 			pinctrl@440000 {
 				cp0_xhci0_vbus_pins: cp0-xhci0-vbus-pins {
diff --git a/arch/arm64/boot/dts/marvell/armada-3900-db.dtsi b/arch/arm64/boot/dts/marvell/armada-3900-db.dtsi
index 230c981..84f8676 100644
--- a/arch/arm64/boot/dts/marvell/armada-3900-db.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-3900-db.dtsi
@@ -89,7 +89,7 @@
 		};
 	};
 
-	cp110-0 {
+	cp0 {
 		config-space {
 			cp0_usb3h0: usb3@500000 {
 				status = "okay";
diff --git a/arch/arm64/boot/dts/marvell/armada-7020-amc.dts b/arch/arm64/boot/dts/marvell/armada-7020-amc.dts
index 0fe4937..761cea9 100644
--- a/arch/arm64/boot/dts/marvell/armada-7020-amc.dts
+++ b/arch/arm64/boot/dts/marvell/armada-7020-amc.dts
@@ -74,7 +74,7 @@
 		};
 	};
 
-	cp110-0 {
+	cp0 {
 		config-space {
 			pinctrl@440000 {
 				i2c1_pins: i2c-pins-1 {
diff --git a/arch/arm64/boot/dts/marvell/armada-7020.dtsi b/arch/arm64/boot/dts/marvell/armada-7020.dtsi
index 3aaf091..ff194e7 100644
--- a/arch/arm64/boot/dts/marvell/armada-7020.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-7020.dtsi
@@ -45,8 +45,50 @@
  * one CP110.
  */
 
+#include "armada-common.dtsi"
+#include <dt-bindings/phy/phy-comphy-mvebu.h>
+
 #include "armada-ap806-dual.dtsi"
-#include "armada-cp110-0.dtsi"
+
+/* CP110-0 Settings */
+#define CP110_NAME		cp0
+#define CP110_NUM		0
+#define CP110_BASE              0xf2000000
+#define CP110_PCIE_MEM_BASE     0xf6000000
+#define CP110_PCIE_MEM_OFFSET	0x1000000
+#define CP110_PCIE_IO_BASE      0xf9000000
+
+#include "armada-cp110.dtsi"
+
+#undef CP110_NAME
+#undef CP110_NUM
+#undef CP110_BASE
+#undef CP110_PCIE_IO_BASE
+#undef CP110_PCIE_MEM_BASE
+#undef CP110_SPI_BUS_ID
+#undef CP110_PCIE_MEM_OFFSET
+#undef CP110_EIP197_INDEX
+
+&cp0_thermal {
+	/* interrupt assignment by ICU supports single thermal sensor unit,
+	* currently cp0 thermal sensor irq only is enabled.
+	* To enable cp1 thermal sensor interrupt,
+	* first disable cp0 thermal sensor interrupt.
+	*/
+	interrupts-extended = <&sei 37>;
+};
+
+&cp0_emac0 {
+	mac-address = [00 00 00 00 00 01];
+};
+
+&cp0_emac2 {
+	mac-address = [00 00 00 00 00 02];
+};
+
+&cp0_emac3 {
+	mac-address = [00 00 00 00 00 03];
+};
 
 / {
 	model = "Marvell Armada 7020";
@@ -80,7 +122,7 @@
 		};
 	};
 
-	cp110-0 {
+	cp0 {
 		config-space {
 			pinctrl@440000 {
 			compatible = "marvell,a70x0-pinctrl";
diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db-B.dts b/arch/arm64/boot/dts/marvell/armada-7040-db-B.dts
index 2f6b6b6..4fa7b3d 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-db-B.dts
+++ b/arch/arm64/boot/dts/marvell/armada-7040-db-B.dts
@@ -76,7 +76,7 @@
 		};
 	};
 
-	cp110-0 {
+	cp0 {
 		config-space {
 			serial@702000 {
 				status = "disabled";
diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db-C.dts b/arch/arm64/boot/dts/marvell/armada-7040-db-C.dts
index d4dba5a..2f100b9 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-db-C.dts
+++ b/arch/arm64/boot/dts/marvell/armada-7040-db-C.dts
@@ -76,7 +76,7 @@
 		};
 	};
 
-	cp110-0 {
+	cp0 {
 		config-space {
 			serial@702000 {
 				status = "disabled";
diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db-D.dts b/arch/arm64/boot/dts/marvell/armada-7040-db-D.dts
index 8525b2d..07ad7ef 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-db-D.dts
+++ b/arch/arm64/boot/dts/marvell/armada-7040-db-D.dts
@@ -76,7 +76,7 @@
 		};
 	};
 
-	cp110-0 {
+	cp0 {
 		config-space {
 			pinctrl@440000 {
 				compatible = "marvell,a70x0-pinctrl";
diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db-E.dts b/arch/arm64/boot/dts/marvell/armada-7040-db-E.dts
index da2e1c9..ac9ff21 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-db-E.dts
+++ b/arch/arm64/boot/dts/marvell/armada-7040-db-E.dts
@@ -71,7 +71,7 @@
 		};
 	};
 
-	cp110-0 {
+	cp0 {
 		config-space {
 			serial@702000 {
 				status = "disabled";
diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db-all-pcie.dts b/arch/arm64/boot/dts/marvell/armada-7040-db-all-pcie.dts
index 0f3e208..4face2f 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-db-all-pcie.dts
+++ b/arch/arm64/boot/dts/marvell/armada-7040-db-all-pcie.dts
@@ -71,7 +71,7 @@
 		};
 	};
 
-	cp110-0 {
+	cp0 {
 		config-space {
 			serial@702000 {
 				status = "disabled";
diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db-nand.dts b/arch/arm64/boot/dts/marvell/armada-7040-db-nand.dts
index 91a77ad..99acb8d 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-db-nand.dts
+++ b/arch/arm64/boot/dts/marvell/armada-7040-db-nand.dts
@@ -71,7 +71,7 @@
 		};
 	};
 
-	cp110-0 {
+	cp0 {
 		config-space {
 			serial@702000 {
 				status = "disabled";
diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db-pp2-vanilla.dts b/arch/arm64/boot/dts/marvell/armada-7040-db-pp2-vanilla.dts
index 1dcc084..1890715 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-db-pp2-vanilla.dts
+++ b/arch/arm64/boot/dts/marvell/armada-7040-db-pp2-vanilla.dts
@@ -77,7 +77,7 @@
 		};
 	};
 
-	cp110-0 {
+	cp0 {
 		config-space {
 			/delete-node/ppv22@000000;
 			serial@702000 {
diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db.dts b/arch/arm64/boot/dts/marvell/armada-7040-db.dts
index 57f31aa..5b40c11 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-db.dts
+++ b/arch/arm64/boot/dts/marvell/armada-7040-db.dts
@@ -71,7 +71,7 @@
 		};
 	};
 
-	cp110-0 {
+	cp0 {
 		config-space {
 			serial@702000 {
 				status = "disabled";
diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi b/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi
index a635021..5f4fe91 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi
@@ -76,7 +76,7 @@
 		};
 	};
 
-	cp110-0 {
+	cp0 {
 		config-space {
 			cp0_i2c0: i2c@701000 {
 				pinctrl-names = "default";
diff --git a/arch/arm64/boot/dts/marvell/armada-7040-pcac.dts b/arch/arm64/boot/dts/marvell/armada-7040-pcac.dts
index 9cb60bc..d678de5 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-pcac.dts
+++ b/arch/arm64/boot/dts/marvell/armada-7040-pcac.dts
@@ -91,7 +91,7 @@
 		};
 	};
 
-	cp110-0 {
+	cp0 {
 		config-space {
 			cp0_comphy: comphy {
 				skip_pcie_power_off;
diff --git a/arch/arm64/boot/dts/marvell/armada-7040.dtsi b/arch/arm64/boot/dts/marvell/armada-7040.dtsi
index 68746bc..d9b4a6f 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-7040.dtsi
@@ -45,8 +45,59 @@
  * one CP110.
  */
 
+#include "armada-common.dtsi"
+#include <dt-bindings/phy/phy-comphy-mvebu.h>
+
 #include "armada-ap806-quad.dtsi"
-#include "armada-cp110-0.dtsi"
+
+/* General settings related tp AP, to be used in CP */
+#define AP_ICU_SET_SPI_HIGH	0x0
+#define AP_ICU_SET_SPI_LOW	0xF03F0040
+#define AP_ICU_CLR_SPI_HIGH	0x0
+#define AP_ICU_CLR_SPI_LOW	0xF03F0048
+
+/* XOR MSI parent for CPs is GIC-v2m in AP */
+#define XOR_MSI_PARENT(XOR_NUM) <&gic_v2m0>
+
+/* CP110-0 Settings */
+#define CP110_NAME		cp0
+#define CP110_NUM		0
+#define CP110_BASE              0xf2000000
+#define CP110_PCIE_MEM_BASE     0xf6000000
+#define CP110_PCIE_MEM_OFFSET	0x1000000
+#define CP110_PCIE_IO_BASE      0xf9000000
+
+#include "armada-cp110.dtsi"
+
+#undef CP110_NAME
+#undef CP110_NUM
+#undef CP110_BASE
+#undef CP110_PCIE_IO_BASE
+#undef CP110_PCIE_MEM_BASE
+#undef CP110_PCIE_MEM_OFFSET
+#undef CP110_SPI_BUS_ID
+#undef CP110_EIP197_INDEX
+
+&cp0_thermal {
+	/* interrupt assignment by ICU supports single thermal sensor unit,
+	* currently cp0 thermal sensor irq only is enabled.
+	* To enable cp1 thermal sensor interrupt,
+	* first disable cp0 thermal sensor interrupt.
+	*/
+	interrupts-extended = <&sei 37>;
+};
+
+&cp0_emac0 {
+	mac-address = [00 00 00 00 00 01];
+};
+
+&cp0_emac2 {
+	mac-address = [00 00 00 00 00 02];
+};
+
+&cp0_emac3 {
+	mac-address = [00 00 00 00 00 03];
+};
 
 / {
 	model = "Marvell Armada 7040";
@@ -79,7 +130,7 @@
 		};
 	};
 
-	cp110-0 {
+	cp0 {
 		config-space {
 			pinctrl@440000 {
 			compatible = "marvell,a70x0-pinctrl";
diff --git a/arch/arm64/boot/dts/marvell/armada-70x0-customer.dts b/arch/arm64/boot/dts/marvell/armada-70x0-customer.dts
index 2b720bc..b26f858 100644
--- a/arch/arm64/boot/dts/marvell/armada-70x0-customer.dts
+++ b/arch/arm64/boot/dts/marvell/armada-70x0-customer.dts
@@ -112,7 +112,7 @@
 		};
 	};
 
-	cp110-0 {
+	cp0 {
 		config-space {
 			pinctrl@440000 {
 				u3d_vbus_pins: pca0_pins {
diff --git a/arch/arm64/boot/dts/marvell/armada-8020.dtsi b/arch/arm64/boot/dts/marvell/armada-8020.dtsi
index 11aeff6..3e5384a 100644
--- a/arch/arm64/boot/dts/marvell/armada-8020.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-8020.dtsi
@@ -45,9 +45,77 @@
  * two CP110.
  */
 
+#include "armada-common.dtsi"
+#include <dt-bindings/phy/phy-comphy-mvebu.h>
+
 #include "armada-ap806-dual.dtsi"
-#include "armada-cp110-0.dtsi"
-#include "armada-cp110-1.dtsi"
+
+/* CP110-0 Settings */
+#define CP110_NAME		cp0
+#define CP110_NUM		0
+#define CP110_BASE              0xf2000000
+#define CP110_PCIE_MEM_BASE     0xf6000000
+#define CP110_PCIE_MEM_OFFSET	0x1000000
+#define CP110_PCIE_IO_BASE      0xf9000000
+
+#include "armada-cp110.dtsi"
+
+#undef CP110_NAME
+#undef CP110_NUM
+#undef CP110_BASE
+#undef CP110_PCIE_IO_BASE
+#undef CP110_PCIE_MEM_OFFSET
+#undef CP110_PCIE_MEM_BASE
+
+&cp0_thermal {
+	/* interrupt assignment by ICU supports single thermal sensor unit,
+	* currently cp0 thermal sensor irq only is enabled.
+	* To enable cp1 thermal sensor interrupt,
+	* first disable cp0 thermal sensor interrupt.
+	*/
+	interrupts-extended = <&sei 37>;
+};
+
+&cp0_emac0 {
+	mac-address = [00 00 00 00 00 01];
+};
+
+&cp0_emac2 {
+	mac-address = [00 00 00 00 00 02];
+};
+
+&cp0_emac3 {
+	mac-address = [00 00 00 00 00 03];
+};
+
+/* CP110-1 Settings */
+#define CP110_NAME		cp1
+#define CP110_NUM		1
+#define CP110_BASE              0xf4000000
+#define CP110_PCIE_MEM_BASE     0xfa000000
+#define CP110_PCIE_MEM_OFFSET	0x1000000
+#define CP110_PCIE_IO_BASE      0xfd000000
+
+#include "armada-cp110.dtsi"
+
+#undef CP110_NAME
+#undef CP110_NUM
+#undef CP110_BASE
+#undef CP110_PCIE_IO_BASE
+#undef CP110_PCIE_MEM_OFFSET
+#undef CP110_PCIE_MEM_BASE
+
+&cp1_emac0 {
+	mac-address = [00 00 00 00 00 04];
+};
+
+&cp1_emac2 {
+	mac-address = [00 00 00 00 00 05];
+};
+
+&cp1_emac3 {
+	mac-address = [00 00 00 00 00 06];
+};
 
 / {
 	model = "Marvell Armada 8020";
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db-B.dts b/arch/arm64/boot/dts/marvell/armada-8040-db-B.dts
index 85dfea6..60f58fc 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db-B.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db-B.dts
@@ -80,7 +80,7 @@
 		};
 	};
 
-	cp110-0 {
+	cp0 {
 		config-space {
 			/* usb3h0 is USB2 only (UTMI) */
 			cp0_usb3h0: usb3@500000 {
@@ -156,7 +156,7 @@
 			status = "okay";
 		};
 	};
-	cp110-1 {
+	cp1 {
 		config-space {
 			cp1_usb3h0: usb3@500000 {
 				separated-phys-for-usb2-usb3;
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db-C.dts b/arch/arm64/boot/dts/marvell/armada-8040-db-C.dts
index 6c6cfcb..c0271fc 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db-C.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db-C.dts
@@ -80,7 +80,7 @@
 		};
 	};
 
-	cp110-0 {
+	cp0 {
 		config-space {
 			sata@540000 {
 				status = "okay";
@@ -170,7 +170,7 @@
 			status = "okay";
 		};
 	};
-	cp110-1 {
+	cp1 {
 		config-space {
 			/* cp1_usb3h0 is USB2 only (UTMI) */
 			cp1_usb3h0: usb3@500000 {
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db-D.dts b/arch/arm64/boot/dts/marvell/armada-8040-db-D.dts
index 7f81e9a..eaee977 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db-D.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db-D.dts
@@ -81,7 +81,7 @@
 		};
 	};
 
-	cp110-0 {
+	cp0 {
 		config-space {
 			sata@540000 {
 				status = "okay";
@@ -171,7 +171,7 @@
 			status = "okay";
 		};
 	};
-	cp110-1 {
+	cp1 {
 		config-space {
 			/* cp1_usb3h0 is USB2 only (UTMI) */
 			cp1_usb3h0: usb3@500000 {
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db-E.dts b/arch/arm64/boot/dts/marvell/armada-8040-db-E.dts
index 68fa453..3756ad4 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db-E.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db-E.dts
@@ -82,7 +82,7 @@
 		};
 	};
 
-	cp110-0 {
+	cp0 {
 		config-space {
 			sata@540000 {
 				status = "okay";
@@ -173,7 +173,7 @@
 			status = "okay";
 		};
 	};
-	cp110-1 {
+	cp1 {
 		config-space {
 			/* cp1_usb3h0 is USB2 only (UTMI) */
 			cp1_usb3h0: usb3@500000 {
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db-pp2-vanilla.dts b/arch/arm64/boot/dts/marvell/armada-8040-db-pp2-vanilla.dts
index 85552b7..6047bca 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db-pp2-vanilla.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db-pp2-vanilla.dts
@@ -81,7 +81,7 @@
 		};
 	};
 
-	cp110-0 {
+	cp0 {
 		config-space {
 			/delete-node/ppv22@000000;
 			sata@540000 {
@@ -180,7 +180,7 @@
 			status = "okay";
 		};
 	};
-	cp110-1 {
+	cp1 {
 		config-space {
 			/delete-node/ppv22@000000;
 			/* cp1_usb3h0 is USB2 only (UTMI) */
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db.dts b/arch/arm64/boot/dts/marvell/armada-8040-db.dts
index 0443cb0..a84b6bb 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db.dts
@@ -81,7 +81,7 @@
 		};
 	};
 
-	cp110-0 {
+	cp0 {
 		config-space {
 			sata@540000 {
 				status = "okay";
@@ -171,7 +171,7 @@
 			status = "okay";
 		};
 	};
-	cp110-1 {
+	cp1 {
 		config-space {
 			/* cp1_usb3h0 is USB2 only (UTMI) */
 			cp1_usb3h0: usb3@500000 {
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi b/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi
index e39a86d..d9de2e0 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi
@@ -90,7 +90,7 @@
 		};
 	};
 
-	cp110-0 {
+	cp0 {
 		config-space {
 			i2c@701000 {
 				status = "okay";
@@ -207,7 +207,7 @@
 		};
 	};
 
-	cp110-1 {
+	cp1 {
 		config-space {
 			i2c@701000 {
 				eeprom@57 {
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-mcbin-pp2-vanilla.dts b/arch/arm64/boot/dts/marvell/armada-8040-mcbin-pp2-vanilla.dts
index be9e2b3..4bea9cc 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-mcbin-pp2-vanilla.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-mcbin-pp2-vanilla.dts
@@ -87,7 +87,7 @@
 		};
 	};
 
-	cp110-0 {
+	cp0 {
 		config-space {
 			/delete-node/ppv22@000000;
 			pinctrl@440000 {
@@ -229,7 +229,7 @@
 		};
 	};
 
-	cp110-1 {
+	cp1 {
 		config-space {
 			/delete-node/ppv22@000000;
 			cp1_comphy1: comphy {
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts b/arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts
index 8f5c0c6..f2e7c9d 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts
@@ -89,7 +89,7 @@
 		};
 	};
 
-	cp110-0 {
+	cp0 {
 		config-space {
 			pinctrl@440000 {
 				xhci0_vbus_pins: xhci0-vbus-pins {
@@ -194,7 +194,7 @@
 		};
 	};
 
-	cp110-1 {
+	cp1 {
 		config-space {
 			cp1_comphy: comphy {
 				status = "okay";
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-ocp.dts b/arch/arm64/boot/dts/marvell/armada-8040-ocp.dts
index c6dcb07..b5c4d97 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-ocp.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-ocp.dts
@@ -87,7 +87,7 @@
 		};
 	};
 
-	cp110-0 {
+	cp0 {
 		config-space {
 			pinctrl@440000 {
 				xhci0_vbus_pins: xhci0-vbus-pins {
@@ -153,7 +153,7 @@
 		};
 	};
 
-	cp110-1 {
+	cp1 {
 		config-space {
 			cp1_comphy: comphy {
 				status = "okay";
diff --git a/arch/arm64/boot/dts/marvell/armada-8040.dtsi b/arch/arm64/boot/dts/marvell/armada-8040.dtsi
index 3c4bf63..9fd80c6 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-8040.dtsi
@@ -45,9 +45,77 @@
  * two CP110.
  */
 
+#include "armada-common.dtsi"
+#include <dt-bindings/phy/phy-comphy-mvebu.h>
+
 #include "armada-ap806-quad.dtsi"
-#include "armada-cp110-0.dtsi"
-#include "armada-cp110-1.dtsi"
+
+/* CP110-0 Settings */
+#define CP110_NAME		cp0
+#define CP110_NUM		0
+#define CP110_BASE              0xf2000000
+#define CP110_PCIE_MEM_BASE     0xf6000000
+#define CP110_PCIE_MEM_OFFSET	0x1000000
+#define CP110_PCIE_IO_BASE      0xf9000000
+
+#include "armada-cp110.dtsi"
+
+#undef CP110_NAME
+#undef CP110_NUM
+#undef CP110_BASE
+#undef CP110_PCIE_IO_BASE
+#undef CP110_PCIE_MEM_BASE
+#undef CP110_PCIE_MEM_OFFSET
+
+&cp0_thermal {
+	/* interrupt assignment by ICU supports single thermal sensor unit,
+	* currently cp0 thermal sensor irq only is enabled.
+	* To enable cp1 thermal sensor interrupt,
+	* first disable cp0 thermal sensor interrupt.
+	*/
+	interrupts-extended = <&sei 37>;
+};
+
+&cp0_emac0 {
+	mac-address = [00 00 00 00 00 01];
+};
+
+&cp0_emac2 {
+	mac-address = [00 00 00 00 00 02];
+};
+
+&cp0_emac3 {
+	mac-address = [00 00 00 00 00 03];
+};
+
+/* CP110-1 Settings */
+#define CP110_NAME		cp1
+#define CP110_NUM		0
+#define CP110_BASE              0xf4000000
+#define CP110_PCIE_MEM_BASE     0xfa000000
+#define CP110_PCIE_MEM_OFFSET	0x1000000
+#define CP110_PCIE_IO_BASE      0xfd000000
+
+#include "armada-cp110.dtsi"
+
+#undef CP110_NAME
+#undef CP110_NUM
+#undef CP110_BASE
+#undef CP110_PCIE_IO_BASE
+#undef CP110_PCIE_MEM_BASE
+#undef CP110_PCIE_MEM_OFFSET
+
+&cp1_emac0 {
+	mac-address = [00 00 00 00 00 04];
+};
+
+&cp1_emac2 {
+	mac-address = [00 00 00 00 00 05];
+};
+
+&cp1_emac3 {
+	mac-address = [00 00 00 00 00 06];
+};
 
 / {
 	model = "Marvell Armada 8040";
@@ -120,7 +188,7 @@
 		};
 	};
 
-	cp110-0 {
+	cp0 {
 		config-space {
 			/* Temp: disable all Coresight nodes */
 			ppv22@000000 {
@@ -177,7 +245,7 @@
 			msi-parent = <&gic_v2m0>;
 		};
 	};
-	cp110-1 {
+	cp1 {
 		config-space {
 			/* Temp: disable all Coresight nodes */
 			ppv22@000000 {
diff --git a/arch/arm64/boot/dts/marvell/armada-80x0-customer.dts b/arch/arm64/boot/dts/marvell/armada-80x0-customer.dts
index 5bce7f8..d356bf6 100644
--- a/arch/arm64/boot/dts/marvell/armada-80x0-customer.dts
+++ b/arch/arm64/boot/dts/marvell/armada-80x0-customer.dts
@@ -128,7 +128,7 @@
 		};
 	};
 
-	cp110-0 {
+	cp0 {
 		config-space {
 			sata@540000 {
 				status = "disabled";
@@ -237,7 +237,7 @@
 		};
 
 	};
-	cp110-1 {
+	cp1 {
 		config-space {
 			cp1_usb3h0: usb3@500000 {
 				status = "disabled";
diff --git a/arch/arm64/boot/dts/marvell/armada-ap806.dtsi b/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
index c0d7cff..a7da40e 100644
--- a/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
@@ -47,6 +47,15 @@
 #include <dt-bindings/interrupt-controller/arm-gic.h>
 #include <dt-bindings/gpio/gpio.h>
 
+/* General settings related tp AP, to be used in CP south-bridges connected to this AP */
+#define AP_ICU_SET_SPI_HIGH	0x0
+#define AP_ICU_SET_SPI_LOW	0xF03F0040
+#define AP_ICU_CLR_SPI_HIGH	0x0
+#define AP_ICU_CLR_SPI_LOW	0xF03F0048
+
+/* GIC-v2m in this AP is the XOR MSI parent for CPs south-bridges connected to this AP */
+#define XOR_MSI_PARENT(XOR_NUM) <&gic_v2m0>
+
 /dts-v1/;
 
 / {
diff --git a/arch/arm64/boot/dts/marvell/armada-cp110-0.dtsi b/arch/arm64/boot/dts/marvell/armada-cp110-0.dtsi
deleted file mode 100644
index f4d574f..0000000
--- a/arch/arm64/boot/dts/marvell/armada-cp110-0.dtsi
+++ /dev/null
@@ -1,68 +0,0 @@
-/****************************************************************************
-* This program is free software: you can redistribute it and/or modify it
-* under the terms of the GNU General Public License as published by the Free
-* Software Foundation, either version 2 of the License, or any later version.
-*
-* This program is distributed in the hope that it will be useful,
-* but WITHOUT ANY WARRANTY; without even the implied warranty of
-* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-* GNU General Public License for more details.
-*
-* You should have received a copy of the GNU General Public License
-* along with this program.  If not, see <http://www.gnu.org/licenses/>.
-* ***************************************************************************
-*/
-
-/*
- * Device Tree file for Marvell Armada CP110-0.
- */
-
-#include "armada-common.dtsi"
-
-#include <dt-bindings/phy/phy-comphy-mvebu.h>
-
-#define CP110_NAME		cp0
-#define CP110_BASE		0xf2000000
-#define CP110_PCIE_IO_BASE	0xf9000000
-#define CP110_PCIE_MEM_BASE	0xf6000000
-#define CP110_SPI_BUS_ID(n)	(0 + (n))
-#define CP110_EIP197_INDEX	0
-
-/ {
-	cp110-0 {
-		#address-cells = <2>;
-		#size-cells = <2>;
-		compatible = "simple-bus";
-		interrupt-parent = <&cp0_icu>;
-		ranges;
-
-		#include "armada-cp110.dtsi"
-	};
-};
-#undef CP110_NAME
-#undef CP110_BASE
-#undef CP110_PCIE_IO_BASE
-#undef CP110_PCIE_MEM_BASE
-#undef CP110_SPI_BUS_ID
-#undef CP110_EIP197_INDEX
-
-&cp0_thermal {
-	/* interrupt assignment by ICU supports single thermal sensor unit,
-	* currently cp0 thermal sensor irq only is enabled.
-	* To enable cp1 thermal sensor interrupt,
-	* first disable cp0 thermal sensor interrupt.
-	*/
-	interrupts-extended = <&sei 37>;
-};
-
-&cp0_emac0 {
-	mac-address = [00 00 00 00 00 01];
-};
-
-&cp0_emac2 {
-	mac-address = [00 00 00 00 00 02];
-};
-
-&cp0_emac3 {
-	mac-address = [00 00 00 00 00 03];
-};
diff --git a/arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi b/arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi
deleted file mode 100644
index 9befadb..0000000
--- a/arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi
+++ /dev/null
@@ -1,59 +0,0 @@
-/****************************************************************************
-* This program is free software: you can redistribute it and/or modify it
-* under the terms of the GNU General Public License as published by the Free
-* Software Foundation, either version 2 of the License, or any later version.
-*
-* This program is distributed in the hope that it will be useful,
-* but WITHOUT ANY WARRANTY; without even the implied warranty of
-* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-* GNU General Public License for more details.
-*
-* You should have received a copy of the GNU General Public License
-* along with this program.  If not, see <http://www.gnu.org/licenses/>.
-* ***************************************************************************
-*/
-
-/*
- * Device Tree file for Marvell Armada CPN-110 Master.
- */
-
-#include "armada-common.dtsi"
-
-#include <dt-bindings/phy/phy-comphy-mvebu.h>
-
-#define CP110_NAME		cp1
-#define CP110_BASE		0xf4000000
-#define CP110_PCIE_IO_BASE	0xfd000000
-#define CP110_PCIE_MEM_BASE	0xfa000000
-#define CP110_SPI_BUS_ID(n)	(2 + (n))
-#define CP110_EIP197_INDEX	1
-
-/ {
-	cp110-1 {
-		#address-cells = <2>;
-		#size-cells = <2>;
-		compatible = "simple-bus";
-		interrupt-parent = <&cp1_icu>;
-		ranges;
-
-		#include "armada-cp110.dtsi"
-	};
-};
-#undef CP110_NAME
-#undef CP110_BASE
-#undef CP110_PCIE_IO_BASE
-#undef CP110_PCIE_MEM_BASE
-#undef CP110_SPI_BUS_ID
-#undef CP110_EIP197_INDEX
-
-&cp1_emac0 {
-	mac-address = [00 00 00 00 00 04];
-};
-
-&cp1_emac2 {
-	mac-address = [00 00 00 00 00 05];
-};
-
-&cp1_emac3 {
-	mac-address = [00 00 00 00 00 06];
-};
diff --git a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
index 38c283b..49a00ec 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
@@ -26,7 +26,7 @@
 #define CP110_PCIEx_IO_BASE_L(iface)	((CP110_PCIE_IO_BASE & 0xffffffff) + (iface *  0x10000))
 
 #define CP110_PCIEx_MEM_BASE_H(iface)	((CP110_PCIE_MEM_BASE >> 32) & 0xffffffff)
-#define CP110_PCIEx_MEM_BASE_L(iface)	((CP110_PCIE_MEM_BASE & 0xffffffff) + (iface *  0x1000000))
+#define CP110_PCIEx_MEM_BASE_L(iface)	((CP110_PCIE_MEM_BASE & 0xffffffff) + (iface *  CP110_PCIE_MEM_OFFSET))
 
 #define CP110_PCIEx_REG0_BASE_H(iface)	(CP110_BASE_HIGH)
 #define CP110_PCIEx_REG0_BASE_L(iface)	(CP110_BASE_LOW + 0x600000 + (iface * 0x20000))
@@ -41,585 +41,601 @@
 #define CP110_PCIE_EP_REG2_BASE_H	(CP110_BASE_HIGH)
 #define CP110_PCIE_EP_REG2_BASE_L	(CP110_BASE_LOW + 0x604000)
 
-config-space {
-	#address-cells = <1>;
-	#size-cells = <1>;
-	compatible = "simple-bus";
-	ranges = <0x0 CP110_BASE_HIGH CP110_BASE_LOW 0x2000000>;
-
-	CP110_LABEL(syscon0): system-controller@440000 {
-		compatible = "marvell,cp110-system-controller0", "syscon";
-		reg = <0x440000 0x1000>;
-		#clock-cells = <2>;
-		core-clock-output-names =
-			CP110_CLK_LABEL(apll), CP110_CLK_LABEL(ppv2-core), CP110_CLK_LABEL(eip),
-			CP110_CLK_LABEL(core), CP110_CLK_LABEL(nand-core), CP110_CLK_LABEL(emmc);
-		gate-clock-output-names =
-			CP110_CLK_LABEL(audio), CP110_CLK_LABEL(communit), CP110_CLK_LABEL(nand),
-			CP110_CLK_LABEL(ppv2), CP110_CLK_LABEL(sdio), CP110_CLK_LABEL(mg-domain),
-			CP110_CLK_LABEL(mg-core), CP110_CLK_LABEL(xor1), CP110_CLK_LABEL(xor0),
-			CP110_CLK_LABEL(gop-dp), "none", CP110_CLK_LABEL(pcie_x10),
-			CP110_CLK_LABEL(pcie_x11), CP110_CLK_LABEL(pcie_x4), CP110_CLK_LABEL(pcie-xor),
-			CP110_CLK_LABEL(sata), CP110_CLK_LABEL(sata-usb), CP110_CLK_LABEL(main),
-			CP110_CLK_LABEL(sd-mmc), "none", "none",
-			CP110_CLK_LABEL(slow-io), CP110_CLK_LABEL(usb3h0), CP110_CLK_LABEL(usb3h1),
-			CP110_CLK_LABEL(usb3dev), CP110_CLK_LABEL(eip150), CP110_CLK_LABEL(eip197);
-	};
-
-	CP110_LABEL(icu): interrupt-controller@1e0000 {
-		compatible = "marvell,icu";
-		reg = <0x1e0000 0x10>;
-		#interrupt-cells = <3>;
-		#size-cells = <1>;
-		#address-cells = <0>;
-		icu-spi-base = <0>;
-		interrupt-controller;
-		interrupt-parent = <&gic>;
-		gicp-spi = <0x0 0xF03F0040 0x0 0xF03F0048>;
-	};
-
-	CP110_LABEL(rtc): rtc@284000 {
-		compatible = "marvell,armada8k-rtc";
-		reg = <0x284000 0x20>, <0x284080 0x24>;
-		reg-names = "rtc", "rtc-soc";
-		#address-cells = <1>;
-		#size-cells = <0>;
-		interrupts = <ICU_GRP_NSR 77 IRQ_TYPE_LEVEL_HIGH>;
-		status = "disabled";
-	};
-
-	CP110_LABEL(pinctrl): pinctrl@440000 {
-		reg = <0x440000 0x20>;
-	};
-
-	CP110_LABEL(gpio0): gpio@440100 {
-		compatible = "marvell,orion-gpio";
-		reg = <0x440100 0x40>;
-		ngpios = <32>;
-		gpiobase = <20>;
-		gpio-controller;
-		#gpio-cells = <2>;
-		interrupt-controller;
-		#interrupt-cells = <2>;
-		interrupts = <ICU_GRP_NSR 86 IRQ_TYPE_LEVEL_HIGH>,
-			   <ICU_GRP_NSR 85 IRQ_TYPE_LEVEL_HIGH>,
-			   <ICU_GRP_NSR 84 IRQ_TYPE_LEVEL_HIGH>,
-			   <ICU_GRP_NSR 83 IRQ_TYPE_LEVEL_HIGH>;
-	};
-
-	CP110_LABEL(gpio1): gpio@440140 {
-		compatible = "marvell,orion-gpio";
-		reg = <0x440140 0x40>;
-		ngpios = <31>;
-		gpiobase = <52>;
-		gpio-controller;
-		#gpio-cells = <2>;
-		interrupt-controller;
-		#interrupt-cells = <2>;
-		interrupts = <ICU_GRP_NSR 82 IRQ_TYPE_LEVEL_HIGH>,
-			   <ICU_GRP_NSR 81 IRQ_TYPE_LEVEL_HIGH>,
-			   <ICU_GRP_NSR 80 IRQ_TYPE_LEVEL_HIGH>,
-			   <ICU_GRP_NSR 79 IRQ_TYPE_LEVEL_HIGH>;
-	};
-
-	CP110_LABEL(tdm): tdm@7a0000 {
-		compatible = "marvell,armada-a8k-tdm";
-		reg = <0x7a0000 0x20000>;
-		reg-names = "tdm_regs";
-		interrupts = <ICU_GRP_NSR 111 IRQ_TYPE_LEVEL_HIGH>,
-			     <ICU_GRP_NSR 112 IRQ_TYPE_LEVEL_HIGH>,
-			     <ICU_GRP_NSR 113 IRQ_TYPE_LEVEL_HIGH>;
-		clocks = <&CP110_LABEL(syscon0) 1 1>;
-		clock-names = "gateclk";
-		status = "disabled";
-	};
-
-	CP110_LABEL(serial0): serial@702000 {
-		compatible = "snps,dw-apb-uart";
-		reg = <0x702000 0x100>;
-		reg-shift = <2>;
-		interrupts = <ICU_GRP_NSR 122 IRQ_TYPE_LEVEL_HIGH>;
-		reg-io-width = <1>;
-		clocks = <&CP110_LABEL(syscon0) 0  3>;
-		status = "disabled";
-	};
-
-	CP110_LABEL(serial1): serial@702100 {
-		compatible = "snps,dw-apb-uart";
-		reg = <0x702100 0x100>;
-		reg-shift = <2>;
-		interrupts = <ICU_GRP_NSR 123 IRQ_TYPE_LEVEL_HIGH>;
-		reg-io-width = <1>;
-		clocks = <&CP110_LABEL(syscon0) 0  3>;
-		status = "disabled";
-	};
-
-	CP110_LABEL(serial2): serial@702200 {
-		compatible = "snps,dw-apb-uart";
-		reg = <0x702200 0x100>;
-		reg-shift = <2>;
-		interrupts = <ICU_GRP_NSR 124 IRQ_TYPE_LEVEL_HIGH>;
-		reg-io-width = <1>;
-		clocks = <&CP110_LABEL(syscon0) 0  3>;
-		status = "disabled";
-	};
-
-	CP110_LABEL(serial3): serial@702300 {
-		compatible = "snps,dw-apb-uart";
-		reg = <0x702300 0x100>;
-		reg-shift = <2>;
-		interrupts = <ICU_GRP_NSR 125 IRQ_TYPE_LEVEL_HIGH>;
-		reg-io-width = <1>;
-		clocks = <&CP110_LABEL(syscon0) 0  3>;
-		status = "disabled";
-	};
-
-	CP110_LABEL(sata): sata@540000 {
-		compatible = "marvell,armada-cp110-ahci";
-		reg = <0x540000 0x30000>;
-		dma-coherent;
-		interrupts = <ICU_GRP_NSR 107 IRQ_TYPE_LEVEL_HIGH>;
-		clocks = <&CP110_LABEL(syscon0) 1 15>;
-		comreset_u = <0x17>;
-		comwake = <0x6>;
-		status = "disabled";
-		#address-cells = <1>;
-		#size-cells = <0>;
-
-		sata-port@0 {
-			reg = <0>;
-			satus = "disabled";
-		};
-		sata-port@1 {
-			reg = <1>;
-			status = "disabled";
+#define CP110_EIP197_INDEX	CP110_NUM
+#define CP110_SPI_BUS_ID(n)	((CP110_NUM * 2) + (n))
+
+/ {
+	CP110_NAME {
+		#address-cells = <2>;
+		#size-cells = <2>;
+		compatible = "simple-bus";
+		interrupt-parent = <&CP110_LABEL(icu)>;
+		ranges;
+
+		config-space {
+			#address-cells = <1>;
+			#size-cells = <1>;
+			compatible = "simple-bus";
+			ranges = <0x0 CP110_BASE_HIGH CP110_BASE_LOW 0x2000000>;
+
+			CP110_LABEL(syscon0): system-controller@440000 {
+				compatible = "marvell,cp110-system-controller0", "syscon";
+				reg = <0x440000 0x1000>;
+				#clock-cells = <2>;
+				core-clock-output-names =
+					CP110_CLK_LABEL(apll), CP110_CLK_LABEL(ppv2-core), CP110_CLK_LABEL(eip),
+					CP110_CLK_LABEL(core), CP110_CLK_LABEL(nand-core), CP110_CLK_LABEL(emmc);
+				gate-clock-output-names =
+					CP110_CLK_LABEL(audio), CP110_CLK_LABEL(communit), CP110_CLK_LABEL(nand),
+					CP110_CLK_LABEL(ppv2), CP110_CLK_LABEL(sdio), CP110_CLK_LABEL(mg-domain),
+					CP110_CLK_LABEL(mg-core), CP110_CLK_LABEL(xor1), CP110_CLK_LABEL(xor0),
+					CP110_CLK_LABEL(gop-dp), "none", CP110_CLK_LABEL(pcie_x10),
+					CP110_CLK_LABEL(pcie_x11), CP110_CLK_LABEL(pcie_x4), CP110_CLK_LABEL(pcie-xor),
+					CP110_CLK_LABEL(sata), CP110_CLK_LABEL(sata-usb), CP110_CLK_LABEL(main),
+					CP110_CLK_LABEL(sd-mmc), "none", "none",
+					CP110_CLK_LABEL(slow-io), CP110_CLK_LABEL(usb3h0), CP110_CLK_LABEL(usb3h1),
+					CP110_CLK_LABEL(usb3dev), CP110_CLK_LABEL(eip150), CP110_CLK_LABEL(eip197);
+			};
+
+			CP110_LABEL(icu): interrupt-controller@1e0000 {
+				compatible = "marvell,icu";
+				reg = <0x1e0000 0x10>;
+				#interrupt-cells = <3>;
+				#size-cells = <1>;
+				#address-cells = <0>;
+				icu-spi-base = <0>;
+				interrupt-controller;
+				interrupt-parent = <&gic>;
+				gicp-spi = <AP_ICU_SET_SPI_HIGH AP_ICU_SET_SPI_LOW AP_ICU_CLR_SPI_HIGH AP_ICU_CLR_SPI_LOW>;
+
+			};
+
+			CP110_LABEL(rtc): rtc@284000 {
+				compatible = "marvell,armada8k-rtc";
+				reg = <0x284000 0x20>, <0x284080 0x24>;
+				reg-names = "rtc", "rtc-soc";
+				#address-cells = <1>;
+				#size-cells = <0>;
+				interrupts = <ICU_GRP_NSR 77 IRQ_TYPE_LEVEL_HIGH>;
+				status = "disabled";
+			};
+
+			CP110_LABEL(pinctrl): pinctrl@440000 {
+				reg = <0x440000 0x20>;
+			};
+
+			CP110_LABEL(gpio0): gpio@440100 {
+				compatible = "marvell,orion-gpio";
+				reg = <0x440100 0x40>;
+				ngpios = <32>;
+				gpiobase = <20>;
+				gpio-controller;
+				#gpio-cells = <2>;
+				interrupt-controller;
+				#interrupt-cells = <2>;
+				interrupts = <ICU_GRP_NSR 86 IRQ_TYPE_LEVEL_HIGH>,
+					   <ICU_GRP_NSR 85 IRQ_TYPE_LEVEL_HIGH>,
+					   <ICU_GRP_NSR 84 IRQ_TYPE_LEVEL_HIGH>,
+					   <ICU_GRP_NSR 83 IRQ_TYPE_LEVEL_HIGH>;
+			};
+
+			CP110_LABEL(gpio1): gpio@440140 {
+				compatible = "marvell,orion-gpio";
+				reg = <0x440140 0x40>;
+				ngpios = <31>;
+				gpiobase = <52>;
+				gpio-controller;
+				#gpio-cells = <2>;
+				interrupt-controller;
+				#interrupt-cells = <2>;
+				interrupts = <ICU_GRP_NSR 82 IRQ_TYPE_LEVEL_HIGH>,
+					   <ICU_GRP_NSR 81 IRQ_TYPE_LEVEL_HIGH>,
+					   <ICU_GRP_NSR 80 IRQ_TYPE_LEVEL_HIGH>,
+					   <ICU_GRP_NSR 79 IRQ_TYPE_LEVEL_HIGH>;
+			};
+
+			CP110_LABEL(tdm): tdm@7a0000 {
+				compatible = "marvell,armada-a8k-tdm";
+				reg = <0x7a0000 0x20000>;
+				reg-names = "tdm_regs";
+				interrupts = <ICU_GRP_NSR 111 IRQ_TYPE_LEVEL_HIGH>,
+					     <ICU_GRP_NSR 112 IRQ_TYPE_LEVEL_HIGH>,
+					     <ICU_GRP_NSR 113 IRQ_TYPE_LEVEL_HIGH>;
+				clocks = <&CP110_LABEL(syscon0) 1 1>;
+				clock-names = "gateclk";
+				status = "disabled";
+			};
+
+			CP110_LABEL(serial0): serial@702000 {
+				compatible = "snps,dw-apb-uart";
+				reg = <0x702000 0x100>;
+				reg-shift = <2>;
+				interrupts = <ICU_GRP_NSR 122 IRQ_TYPE_LEVEL_HIGH>;
+				reg-io-width = <1>;
+				clocks = <&CP110_LABEL(syscon0) 0  3>;
+				status = "disabled";
+			};
+
+			CP110_LABEL(serial1): serial@702100 {
+				compatible = "snps,dw-apb-uart";
+				reg = <0x702100 0x100>;
+				reg-shift = <2>;
+				interrupts = <ICU_GRP_NSR 123 IRQ_TYPE_LEVEL_HIGH>;
+				reg-io-width = <1>;
+				clocks = <&CP110_LABEL(syscon0) 0  3>;
+				status = "disabled";
+			};
+
+			CP110_LABEL(serial2): serial@702200 {
+				compatible = "snps,dw-apb-uart";
+				reg = <0x702200 0x100>;
+				reg-shift = <2>;
+				interrupts = <ICU_GRP_NSR 124 IRQ_TYPE_LEVEL_HIGH>;
+				reg-io-width = <1>;
+				clocks = <&CP110_LABEL(syscon0) 0  3>;
+				status = "disabled";
+			};
+
+			CP110_LABEL(serial3): serial@702300 {
+				compatible = "snps,dw-apb-uart";
+				reg = <0x702300 0x100>;
+				reg-shift = <2>;
+				interrupts = <ICU_GRP_NSR 125 IRQ_TYPE_LEVEL_HIGH>;
+				reg-io-width = <1>;
+				clocks = <&CP110_LABEL(syscon0) 0  3>;
+				status = "disabled";
+			};
+
+			CP110_LABEL(sata): sata@540000 {
+				compatible = "marvell,armada-cp110-ahci";
+				reg = <0x540000 0x30000>;
+				dma-coherent;
+				interrupts = <ICU_GRP_NSR 107 IRQ_TYPE_LEVEL_HIGH>;
+				clocks = <&CP110_LABEL(syscon0) 1 15>;
+				comreset_u = <0x17>;
+				comwake = <0x6>;
+				status = "disabled";
+				#address-cells = <1>;
+				#size-cells = <0>;
+
+				sata-port@0 {
+					reg = <0>;
+					satus = "disabled";
+				};
+				sata-port@1 {
+					reg = <1>;
+					status = "disabled";
+				};
+			};
+
+			CP110_LABEL(usb3h0): usb3@500000 {
+				compatible = "generic-xhci";
+				reg = <0x500000 0x4000>;
+				dma-coherent;
+				interrupts = <ICU_GRP_NSR 106 IRQ_TYPE_LEVEL_HIGH>;
+				clocks = <&CP110_LABEL(syscon0) 1 22>;
+				needs-reset-on-resume;
+				status = "disabled";
+			};
+
+			CP110_LABEL(usb3h1): usb3@510000 {
+				compatible = "generic-xhci";
+				reg = <0x510000 0x4000>;
+				dma-coherent;
+				interrupts = <ICU_GRP_NSR 105 IRQ_TYPE_LEVEL_HIGH>;
+				clocks = <&CP110_LABEL(syscon0) 1 23>;
+				needs-reset-on-resume;
+				status = "disabled";
+			};
+
+			CP110_LABEL(usb3d): u3d@520000 {
+				compatible = "marvell,mvebu-u3d";
+				/* 0: 0x520000: USB 3.0 Device port 0: DEV_INFO_REG(0:15 - version_id)
+				 * 1: 0x121F10: Clock Source High reg: VBUS present indication in PHY SerDes lane#1
+				 */
+				reg = <0x520000 0x2000>, <0x121f10 0x40>;
+				interrupts = <ICU_GRP_NSR 102 IRQ_TYPE_LEVEL_HIGH>;
+				clocks = <&CP110_LABEL(syscon0) 1 24>;
+				dma-coherent;
+				status = "disabled";
+			};
+
+			CP110_LABEL(udc): udc@524100 {
+				compatible = "marvell,mv-udc";
+				reg = <0x524100 0x2000>;
+				interrupts = <ICU_GRP_NSR 102 IRQ_TYPE_LEVEL_HIGH>;
+				clocks = <&CP110_LABEL(syscon0) 1 24>;
+				dma-coherent;
+				status = "disabled";
+			};
+
+			CP110_LABEL(utmi0): utmi@580000 {
+				compatible = "marvell,cp110-utmi-phy";
+				reg = <0x580000 0x1000>,
+				      <0x440420 0x4>,
+				      <0x440440 0x4>;
+				reg-names = "utmi", "usb-cfg-reg", "utmi-cfg-reg";
+				utmi-port = <UTMI_PHY_TO_USB3_HOST0>;
+				utmi-index = <0>;
+				#phy-cells = <0>;
+				status = "disabled";
+			};
+
+			CP110_LABEL(utmi1): utmi@581000 {
+				compatible = "marvell,cp110-utmi-phy";
+				reg = <0x581000 0x1000>,
+				      <0x440420 0x4>,
+				      <0x440444 0x4>;
+				reg-names = "utmi", "usb-cfg-reg", "utmi-cfg-reg";
+				utmi-port = <UTMI_PHY_TO_USB3_HOST1>;
+				utmi-index = <1>;
+				#phy-cells = <0>;
+				status = "disabled";
+			};
+
+			CP110_LABEL(spi0): spi@700600 {
+				compatible = "marvell,armada-380-spi";
+				reg = <0x700600 0x50>;
+				#address-cells = <0x1>;
+				#size-cells = <0x0>;
+				cell-index = <CP110_SPI_BUS_ID(1)>;
+				clocks = <&CP110_LABEL(syscon0) 1 21>;
+				status = "disabled";
+			};
+
+			CP110_LABEL(spi1): spi@700680 {
+				compatible = "marvell,armada-380-spi";
+				reg = <0x700680 0x50>;
+				#address-cells = <0x1>;
+				#size-cells = <0x0>;
+				cell-index = <CP110_SPI_BUS_ID(2)>;
+				clocks = <&CP110_LABEL(syscon0) 1 21>;
+				status = "disabled";
+			};
+
+			CP110_LABEL(i2c0): i2c@701000 {
+				compatible = "marvell,mv78230-i2c";
+				reg = <0x701000 0x20>;
+				#address-cells = <1>;
+				#size-cells = <0>;
+				interrupts = <ICU_GRP_NSR 120 IRQ_TYPE_LEVEL_HIGH>;
+				timeout-ms = <1000>;
+				clocks = <&CP110_LABEL(syscon0) 1 21>;
+				status = "disabled";
+			};
+
+			CP110_LABEL(i2c1): i2c@701100 {
+				compatible = "marvell,mv78230-i2c";
+				reg = <0x701100 0x20>;
+				#address-cells = <1>;
+				#size-cells = <0>;
+				interrupts = <ICU_GRP_NSR 121 IRQ_TYPE_LEVEL_HIGH>;
+				timeout-ms = <1000>;
+				clocks = <&CP110_LABEL(syscon0) 1 21>;
+				status = "disabled";
+			};
+
+			CP110_LABEL(nand): nand@720000 {
+				compatible = "marvell,armada370-nand";
+				reg = <0x720000 0x54>;
+				#address-cells = <1>;
+				#size-cells = <1>;
+				num-cs = <1>;
+				marvell,nand-enable-arbiter;
+				nand-on-flash-bbt;
+				interrupts = <ICU_GRP_NSR 115 IRQ_TYPE_LEVEL_HIGH>;
+				clocks = <&CP110_LABEL(syscon0) 1 2>;
+				nand-ecc-strength = <4>;
+				nand-ecc-step-size = <512>;
+				status = "disabled";
+			};
+
+			CP110_LABEL(xor0): dma_xor@6a0000 {
+				compatible = "marvell,mv-xor-v2";
+				reg = <0x6a0000 0x1000>,
+					<0x6b0000 0x1000>;
+				dma-coherent;
+				dma-bus-width = <40>;
+				msi-parent = XOR_MSI_PARENT(0);
+				clocks = <&CP110_LABEL(syscon0) 1 8>;
+				status = "okay";
+			};
+
+			CP110_LABEL(xor1): dma_xor@6c0000 {
+				compatible = "marvell,mv-xor-v2";
+				reg = <0x6c0000 0x1000>,
+					<0x6d0000 0x1000>;
+				dma-coherent;
+				dma-bus-width = <40>;
+				msi-parent = XOR_MSI_PARENT(1);
+				clocks = <&CP110_LABEL(syscon0) 1 7>;
+				status = "okay";
+			};
+
+			CP110_LABEL(thermal): thermal@400078 {
+				compatible = "marvell,armada-cp110-thermal";
+				reg = <0x400078 0x4>,
+					<0x400070 0x8>,
+					<0x400100 0x20>;
+				threshold = <100>;
+				hysteresis = <2>;
+				status = "okay";
+			};
+
+			CP110_LABEL(mdio0): mdio@12a200 {
+				#address-cells = <1>;
+				#size-cells = <0>;
+				compatible = "marvell,orion-mdio";
+				reg = <0x12a200 0x10>;
+			};
+
+			CP110_LABEL(mdio1): mdio@15b000 {
+				#address-cells = <1>;
+				#size-cells = <0>;
+				compatible = "marvell,xmdio";
+				reg = <0x12a600 0x10>;
+			};
+
+			CP110_LABEL(sdhci): sdhci@780000 {
+				compatible = "marvell,armada-cp110-sdhci";
+				reg = <0x780000 0x300>;
+				interrupts = <ICU_GRP_NSR 27 IRQ_TYPE_LEVEL_HIGH>;
+				clock-names = "core";
+				clocks = <&CP110_LABEL(syscon0) 1 4>;
+				dma-coherent;
+				marvell,xenon-phy-tun-step-divider = <25>;
+				status = "disabled";
+			};
+
+			CP110_LABEL(axim_cp_rd): axim-cp-rd@3c5000 {
+				compatible = "marvell,coresight-axim", "arm,primecell";
+				reg = <0x3c5000 0x1000>;
+				clocks = <&syscon 3>;
+				clock-names = "apb_pclk";
+				bus-width = <40>;
+			};
+
+			CP110_LABEL(axim_cp_wr): axim-cp-wr@3c6000 {
+				compatible = "marvell,coresight-axim", "arm,primecell";
+				reg = <0x3c6000 0x1000>;
+				clocks = <&syscon 3>;
+				clock-names = "apb_pclk";
+				bus-width = <40>;
+			};
+
+			CP110_LABEL(axim_ppv2_rd): axim-ppv2-rd@3c0000 {
+				compatible = "marvell,coresight-axim", "arm,primecell";
+				reg = <0x3c0000 0x1000>;
+				clocks = <&syscon 3>;
+				clock-names = "apb_pclk";
+				bus-width = <40>;
+			};
+
+			CP110_LABEL(axim_ppv2_wr): axim-ppv2-wr@3c1000 {
+				compatible = "marvell,coresight-axim", "arm,primecell";
+				reg = <0x3c1000 0x1000>;
+				clocks = <&syscon 3>;
+				clock-names = "apb_pclk";
+				bus-width = <40>;
+			};
+
+			CP110_LABEL(axim_hb1_rd): axim-hb1-rd@3c8000 {
+				compatible = "marvell,coresight-axim", "arm,primecell";
+				reg = <0x3c8000 0x1000>;
+				clocks = <&syscon 3>;
+				clock-names = "apb_pclk";
+				bus-width = <40>;
+			};
+
+			CP110_LABEL(axim_hb1_wr): axim-hb1-wr@3c9000 {
+				compatible = "marvell,coresight-axim", "arm,primecell";
+				reg = <0x3c9000 0x1000>;
+				clocks = <&syscon 3>;
+				clock-names = "apb_pclk";
+				bus-width = <40>;
+			};
+
+			gop {
+				CP110_LABEL(emac0): mac0 {
+					interrupts = <ICU_GRP_NSR 129 IRQ_TYPE_LEVEL_HIGH>;  /* Link IRQ */
+					port-id = <0>; /* gop_port_id */
+				};
+				CP110_LABEL(emac2): mac2 {
+					interrupts = <ICU_GRP_NSR 128 IRQ_TYPE_LEVEL_HIGH>;  /* Link IRQ */
+					port-id = <2>; /* gop_port_id */
+				};
+				CP110_LABEL(emac3): mac3 {
+					interrupts = <ICU_GRP_NSR 127 IRQ_TYPE_LEVEL_HIGH>;  /* Link IRQ */
+					port-id = <3>; /* gop_port_id */
+				};
+			};
+
+			CP110_LABEL(ppv22): ppv22@000000 {
+				compatible = "marvell,mv-pp22";
+				reg = <0x000000 0x90000>, /* Packet Processor regs */
+					<0x129000 0x0600>, /* XMIB regs */
+					<0x12a000 0x200>,  /* LED regs */
+					<0x12a200 0x200>,  /* SMI regs */
+					<0x12a400 0x200>,  /* TAI regs */
+					<0x12a600 0x200>,  /* XSMI regs */
+					<0x12b000 0x1000>,  /* MG Internal regs */
+					<0x130000 0x6000>, /* MSPG regs */
+					<0x130400 0x200>,  /* MSPG - XPCS regs */
+					<0x130600 0x200>,  /* FCA - flow control regs*/
+					<0x130e00 0x100>,  /* MSPG - GMAC regs */
+					<0x130f00 0x100>,  /* MSPG - XLG MAC regs */
+					<0x441100 0x100>;  /* RFU-1 Regs */
+				reg-names = "pp", "xmib", "led", "smi", "tai", "xsmi", "mg", "mspg", "xpcs",
+						"fca", "gmac", "xlg", "rfu1";
+				clocks = <&CP110_LABEL(syscon0) 1 3>, <&CP110_LABEL(syscon0) 1 18>, <&CP110_LABEL(syscon0) 1 9>,
+					<&CP110_LABEL(syscon0) 1 6>, <&CP110_LABEL(syscon0) 1 5>;
+				clock-names = "pp_clk", "gop_core_clk", "gop_clk", "mg_core_clk", "mg_clk";
+				status = "okay";
+				CP110_LABEL(eth0): eth0@010000 {
+					interrupts = <ICU_GRP_NSR 39 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#0 Interrupt */
+							<ICU_GRP_NSR 43 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#1 Interrupt */
+							<ICU_GRP_NSR 47 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#2 Interrupt */
+							<ICU_GRP_NSR 51 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#3 Interrupt */
+							<ICU_GRP_NSR 55 IRQ_TYPE_LEVEL_HIGH>; /* RX Shared Interrupt */
+					port-id = <0>; /* pp2_port_id */
+					emac-data = <&CP110_LABEL(emac0)>;
+					status = "disabled";
+				};
+				CP110_LABEL(eth1): eth1@020000 {
+					interrupts = <ICU_GRP_NSR 40 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#0 Interrupt */
+							<ICU_GRP_NSR 44 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#1 Interrupt */
+							<ICU_GRP_NSR 48 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#2 Interrupt */
+							<ICU_GRP_NSR 52 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#3 Interrupt */
+							<ICU_GRP_NSR 56 IRQ_TYPE_LEVEL_HIGH>; /* RX Shared Interrupt */
+					port-id = <1>; /* pp2_port_id */
+					emac-data = <&CP110_LABEL(emac2)>;
+					status = "disabled";
+				};
+				CP110_LABEL(eth2): eth2@030000 {
+					interrupts = <ICU_GRP_NSR 41 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#0 Interrupt */
+							<ICU_GRP_NSR 45 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#1 Interrupt */
+							<ICU_GRP_NSR 49 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#2 Interrupt */
+							<ICU_GRP_NSR 53 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#3 Interrupt */
+							<ICU_GRP_NSR 57 IRQ_TYPE_LEVEL_HIGH>; /* RX Shared Interrupt */
+
+					port-id = <2>; /* pp2_port_id */
+					emac-data = <&CP110_LABEL(emac3)>;
+					status = "disabled";
+				};
+			};
+
+			CP110_LABEL(trng): trng@760000 {
+				compatible = "marvell,armada-8k-rng", "inside-secure,safexcel-eip76";
+				reg = <0x760000 0x7D>;
+				#address-cells = <1>;
+				#size-cells = <0>;
+				interrupts = <ICU_GRP_NSR 95 IRQ_TYPE_LEVEL_HIGH>;
+				clocks = <&CP110_LABEL(syscon0) 1 25>;
+				status = "okay";
+			};
+
+			CP110_LABEL(eip197): eip197@800000 {
+				compatible = "inside-secure,safexcel-eip197";
+				reg = <0x800000 0x200000>;
+				dma-coherent;
+				dma-bus-width = <40>;
+				interrupts = <ICU_GRP_REI 87 IRQ_TYPE_LEVEL_HIGH>,
+						<ICU_GRP_NSR 88 IRQ_TYPE_LEVEL_HIGH>,
+						<ICU_GRP_NSR 89 IRQ_TYPE_LEVEL_HIGH>,
+						<ICU_GRP_NSR 90 IRQ_TYPE_LEVEL_HIGH>,
+						<ICU_GRP_NSR 91 IRQ_TYPE_LEVEL_HIGH>,
+						<ICU_GRP_NSR 92 IRQ_TYPE_LEVEL_HIGH>;
+				interrupt-names = "eip_addr", "ring0", "ring1",
+						"ring2", "ring3", "eip_out";
+				clocks = <&CP110_LABEL(syscon0) 1 26>;
+				cell-index = <CP110_EIP197_INDEX>;
+				status = "okay";
+			};
+
+			CP110_LABEL(comphy): comphy {
+				compatible = "marvell,cp110-comphy";
+				reg = <0x441000 0x100>, <0x120000 0x6000>;
+				reg-names = "comphy", "serdes";
+				dfx-reg = <0x400280 4>;
+				#phy-cells = <2>;
+				sar-data = <&CP110_LABEL(sar)>;
+				skip_pcie_power_off;
+				status = "disabled";
+			};
+
+			CP110_LABEL(sar): sar@400200 {
+				compatible = "marvell,sample-at-reset-cp110";
+				reg = <0x400200 0x8>;
+				status = "okay";
+			};
 		};
-	};
-
-	CP110_LABEL(usb3h0): usb3@500000 {
-		compatible = "generic-xhci";
-		reg = <0x500000 0x4000>;
-		dma-coherent;
-		interrupts = <ICU_GRP_NSR 106 IRQ_TYPE_LEVEL_HIGH>;
-		clocks = <&CP110_LABEL(syscon0) 1 22>;
-		needs-reset-on-resume;
-		status = "disabled";
-	};
-
-	CP110_LABEL(usb3h1): usb3@510000 {
-		compatible = "generic-xhci";
-		reg = <0x510000 0x4000>;
-		dma-coherent;
-		interrupts = <ICU_GRP_NSR 105 IRQ_TYPE_LEVEL_HIGH>;
-		clocks = <&CP110_LABEL(syscon0) 1 23>;
-		needs-reset-on-resume;
-		status = "disabled";
-	};
-
-	CP110_LABEL(usb3d): u3d@520000 {
-		compatible = "marvell,mvebu-u3d";
-		/* 0: 0x520000: USB 3.0 Device port 0: DEV_INFO_REG(0:15 - version_id)
-		 * 1: 0x121F10: Clock Source High reg: VBUS present indication in PHY SerDes lane#1
-		 */
-		reg = <0x520000 0x2000>, <0x121f10 0x40>;
-		interrupts = <ICU_GRP_NSR 102 IRQ_TYPE_LEVEL_HIGH>;
-		clocks = <&CP110_LABEL(syscon0) 1 24>;
-		dma-coherent;
-		status = "disabled";
-	};
-
-	CP110_LABEL(udc): udc@524100 {
-		compatible = "marvell,mv-udc";
-		reg = <0x524100 0x2000>;
-		interrupts = <ICU_GRP_NSR 102 IRQ_TYPE_LEVEL_HIGH>;
-		clocks = <&CP110_LABEL(syscon0) 1 24>;
-		dma-coherent;
-		status = "disabled";
-	};
-
-	CP110_LABEL(utmi0): utmi@580000 {
-		compatible = "marvell,cp110-utmi-phy";
-		reg = <0x580000 0x1000>,
-		      <0x440420 0x4>,
-		      <0x440440 0x4>;
-		reg-names = "utmi", "usb-cfg-reg", "utmi-cfg-reg";
-		utmi-port = <UTMI_PHY_TO_USB3_HOST0>;
-		utmi-index = <0>;
-		#phy-cells = <0>;
-		status = "disabled";
-	};
-
-	CP110_LABEL(utmi1): utmi@581000 {
-		compatible = "marvell,cp110-utmi-phy";
-		reg = <0x581000 0x1000>,
-		      <0x440420 0x4>,
-		      <0x440444 0x4>;
-		reg-names = "utmi", "usb-cfg-reg", "utmi-cfg-reg";
-		utmi-port = <UTMI_PHY_TO_USB3_HOST1>;
-		utmi-index = <1>;
-		#phy-cells = <0>;
-		status = "disabled";
-	};
-
-	CP110_LABEL(spi0): spi@700600 {
-		compatible = "marvell,armada-380-spi";
-		reg = <0x700600 0x50>;
-		#address-cells = <0x1>;
-		#size-cells = <0x0>;
-		cell-index = <CP110_SPI_BUS_ID(1)>;
-		clocks = <&CP110_LABEL(syscon0) 1 21>;
-		status = "disabled";
-	};
-
-	CP110_LABEL(spi1): spi@700680 {
-		compatible = "marvell,armada-380-spi";
-		reg = <0x700680 0x50>;
-		#address-cells = <0x1>;
-		#size-cells = <0x0>;
-		cell-index = <CP110_SPI_BUS_ID(2)>;
-		clocks = <&CP110_LABEL(syscon0) 1 21>;
-		status = "disabled";
-	};
-
-	CP110_LABEL(i2c0): i2c@701000 {
-		compatible = "marvell,mv78230-i2c";
-		reg = <0x701000 0x20>;
-		#address-cells = <1>;
-		#size-cells = <0>;
-		interrupts = <ICU_GRP_NSR 120 IRQ_TYPE_LEVEL_HIGH>;
-		timeout-ms = <1000>;
-		clocks = <&CP110_LABEL(syscon0) 1 21>;
-		status = "disabled";
-	};
-
-	CP110_LABEL(i2c1): i2c@701100 {
-		compatible = "marvell,mv78230-i2c";
-		reg = <0x701100 0x20>;
-		#address-cells = <1>;
-		#size-cells = <0>;
-		interrupts = <ICU_GRP_NSR 121 IRQ_TYPE_LEVEL_HIGH>;
-		timeout-ms = <1000>;
-		clocks = <&CP110_LABEL(syscon0) 1 21>;
-		status = "disabled";
-	};
-
-	CP110_LABEL(nand): nand@720000 {
-		compatible = "marvell,armada370-nand";
-		reg = <0x720000 0x54>;
-		#address-cells = <1>;
-		#size-cells = <1>;
-		num-cs = <1>;
-		marvell,nand-enable-arbiter;
-		nand-on-flash-bbt;
-		interrupts = <ICU_GRP_NSR 115 IRQ_TYPE_LEVEL_HIGH>;
-		clocks = <&CP110_LABEL(syscon0) 1 2>;
-		nand-ecc-strength = <4>;
-		nand-ecc-step-size = <512>;
-		status = "disabled";
-	};
-
-	CP110_LABEL(xor0): dma_xor@6a0000 {
-		compatible = "marvell,mv-xor-v2";
-		reg = <0x6a0000 0x1000>,
-			<0x6b0000 0x1000>;
-		dma-coherent;
-		dma-bus-width = <40>;
-		msi-parent = <&gic_v2m0>;
-		clocks = <&CP110_LABEL(syscon0) 1 8>;
-		status = "okay";
-	};
-
-	CP110_LABEL(xor1): dma_xor@6c0000 {
-		compatible = "marvell,mv-xor-v2";
-		reg = <0x6c0000 0x1000>,
-			<0x6d0000 0x1000>;
-		dma-coherent;
-		dma-bus-width = <40>;
-		msi-parent = <&gic_v2m0>;
-		clocks = <&CP110_LABEL(syscon0) 1 7>;
-		status = "okay";
-	};
-
-	CP110_LABEL(thermal): thermal@400078 {
-		compatible = "marvell,armada-cp110-thermal";
-		reg = <0x400078 0x4>,
-			<0x400070 0x8>,
-			<0x400100 0x20>;
-		threshold = <100>;
-		hysteresis = <2>;
-		status = "okay";
-	};
-
-	CP110_LABEL(mdio0): mdio@12a200 {
-		#address-cells = <1>;
-		#size-cells = <0>;
-		compatible = "marvell,orion-mdio";
-		reg = <0x12a200 0x10>;
-	};
-
-	CP110_LABEL(mdio1): mdio@15b000 {
-		#address-cells = <1>;
-		#size-cells = <0>;
-		compatible = "marvell,xmdio";
-		reg = <0x12a600 0x10>;
-	};
-
-	CP110_LABEL(sdhci): sdhci@780000 {
-		compatible = "marvell,armada-cp110-sdhci";
-		reg = <0x780000 0x300>;
-		interrupts = <ICU_GRP_NSR 27 IRQ_TYPE_LEVEL_HIGH>;
-		clock-names = "core";
-		clocks = <&CP110_LABEL(syscon0) 1 4>;
-		dma-coherent;
-		marvell,xenon-phy-tun-step-divider = <25>;
-		status = "disabled";
-	};
-
-	CP110_LABEL(axim_cp_rd): axim-cp-rd@3c5000 {
-		compatible = "marvell,coresight-axim", "arm,primecell";
-		reg = <0x3c5000 0x1000>;
-		clocks = <&syscon 3>;
-		clock-names = "apb_pclk";
-		bus-width = <40>;
-	};
-
-	CP110_LABEL(axim_cp_wr): axim-cp-wr@3c6000 {
-		compatible = "marvell,coresight-axim", "arm,primecell";
-		reg = <0x3c6000 0x1000>;
-		clocks = <&syscon 3>;
-		clock-names = "apb_pclk";
-		bus-width = <40>;
-	};
-
-	CP110_LABEL(axim_ppv2_rd): axim-ppv2-rd@3c0000 {
-		compatible = "marvell,coresight-axim", "arm,primecell";
-		reg = <0x3c0000 0x1000>;
-		clocks = <&syscon 3>;
-		clock-names = "apb_pclk";
-		bus-width = <40>;
-	};
-
-	CP110_LABEL(axim_ppv2_wr): axim-ppv2-wr@3c1000 {
-		compatible = "marvell,coresight-axim", "arm,primecell";
-		reg = <0x3c1000 0x1000>;
-		clocks = <&syscon 3>;
-		clock-names = "apb_pclk";
-		bus-width = <40>;
-	};
-
-	CP110_LABEL(axim_hb1_rd): axim-hb1-rd@3c8000 {
-		compatible = "marvell,coresight-axim", "arm,primecell";
-		reg = <0x3c8000 0x1000>;
-		clocks = <&syscon 3>;
-		clock-names = "apb_pclk";
-		bus-width = <40>;
-	};
 
-	CP110_LABEL(axim_hb1_wr): axim-hb1-wr@3c9000 {
-		compatible = "marvell,coresight-axim", "arm,primecell";
-		reg = <0x3c9000 0x1000>;
-		clocks = <&syscon 3>;
-		clock-names = "apb_pclk";
-		bus-width = <40>;
-	};
-
-	gop {
-		CP110_LABEL(emac0): mac0 {
-			interrupts = <ICU_GRP_NSR 129 IRQ_TYPE_LEVEL_HIGH>;  /* Link IRQ */
-			port-id = <0>; /* gop_port_id */
-		};
-		CP110_LABEL(emac2): mac2 {
-			interrupts = <ICU_GRP_NSR 128 IRQ_TYPE_LEVEL_HIGH>;  /* Link IRQ */
-			port-id = <2>; /* gop_port_id */
-		};
-		CP110_LABEL(emac3): mac3 {
-			interrupts = <ICU_GRP_NSR 127 IRQ_TYPE_LEVEL_HIGH>;  /* Link IRQ */
-			port-id = <3>; /* gop_port_id */
+		CP110_LABEL(pcie_ep): pcie-ep@600000 {
+			compatible = "marvell,armada8k-pcie-ep", "snps,dw-pcie";
+			reg = <CP110_PCIE_EP_REG0_BASE_H CP110_PCIE_EP_REG0_BASE_L 0 0x4000>,
+			      <CP110_PCIE_EP_REG1_BASE_H CP110_PCIE_EP_REG1_BASE_L 0 0x80000>,
+			      <CP110_PCIE_EP_REG2_BASE_H CP110_PCIE_EP_REG2_BASE_L 0 0x1000>;
+			reg-names = "core", "lm", "shadow_core";
+			clocks = <&CP110_LABEL(syscon0) 1 13>;
+			status = "disabled";
 		};
-	};
 
-	CP110_LABEL(ppv22): ppv22@000000 {
-		compatible = "marvell,mv-pp22";
-		reg = <0x000000 0x90000>, /* Packet Processor regs */
-			<0x129000 0x0600>, /* XMIB regs */
-			<0x12a000 0x200>,  /* LED regs */
-			<0x12a200 0x200>,  /* SMI regs */
-			<0x12a400 0x200>,  /* TAI regs */
-			<0x12a600 0x200>,  /* XSMI regs */
-			<0x12b000 0x1000>,  /* MG Internal regs */
-			<0x130000 0x6000>, /* MSPG regs */
-			<0x130400 0x200>,  /* MSPG - XPCS regs */
-			<0x130600 0x200>,  /* FCA - flow control regs*/
-			<0x130e00 0x100>,  /* MSPG - GMAC regs */
-			<0x130f00 0x100>,  /* MSPG - XLG MAC regs */
-			<0x441100 0x100>;  /* RFU-1 Regs */
-		reg-names = "pp", "xmib", "led", "smi", "tai", "xsmi", "mg", "mspg", "xpcs",
-				"fca", "gmac", "xlg", "rfu1";
-		clocks = <&CP110_LABEL(syscon0) 1 3>, <&CP110_LABEL(syscon0) 1 18>, <&CP110_LABEL(syscon0) 1 9>,
-			<&CP110_LABEL(syscon0) 1 6>, <&CP110_LABEL(syscon0) 1 5>;
-		clock-names = "pp_clk", "gop_core_clk", "gop_clk", "mg_core_clk", "mg_clk";
-		status = "okay";
-		CP110_LABEL(eth0): eth0@010000 {
-			interrupts = <ICU_GRP_NSR 39 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#0 Interrupt */
-					<ICU_GRP_NSR 43 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#1 Interrupt */
-					<ICU_GRP_NSR 47 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#2 Interrupt */
-					<ICU_GRP_NSR 51 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#3 Interrupt */
-					<ICU_GRP_NSR 55 IRQ_TYPE_LEVEL_HIGH>; /* RX Shared Interrupt */
-			port-id = <0>; /* pp2_port_id */
-			emac-data = <&CP110_LABEL(emac0)>;
+		CP110_LABEL(pcie0): pcie0@600000 {
+			compatible = "marvell,armada8k-pcie", "snps,dw-pcie";
+			reg = <CP110_PCIEx_REG0_BASE_H(0) CP110_PCIEx_REG0_BASE_L(0) 0 0x10000>,
+			      <CP110_PCIEx_REG1_BASE_H(0) CP110_PCIEx_REG1_BASE_L(0) 0 0x80000>;	/* Last 512KB of mem space */
+			reg-names = "ctrl", "config";
+			#address-cells = <3>;
+			#size-cells = <2>;
+			#interrupt-cells = <1>;
+			device_type = "pci";
+			dma-coherent;
+
+			bus-range = <0 0xff>;
+			ranges =
+				/* downstream I/O */
+				<0x81000000 0 CP110_PCIEx_IO_BASE_L(0) CP110_PCIEx_IO_BASE_H(0) CP110_PCIEx_IO_BASE_L(0) 0 0x10000
+				/* non-prefetchable memory */
+				 0x82000000 0 CP110_PCIEx_MEM_BASE_L(0) CP110_PCIEx_MEM_BASE_H(0) CP110_PCIEx_MEM_BASE_L(0) 0 0xf00000>;
+			interrupt-map-mask = <0 0 0 0>;
+			interrupt-map = <0 0 0 0 &CP110_LABEL(icu) ICU_GRP_NSR 22 IRQ_TYPE_LEVEL_HIGH>;
+			interrupts = <ICU_GRP_NSR 22 IRQ_TYPE_LEVEL_HIGH>;
+			num-lanes = <1>;
+			clocks = <&CP110_LABEL(syscon0) 1 13>;
 			status = "disabled";
 		};
-		CP110_LABEL(eth1): eth1@020000 {
-			interrupts = <ICU_GRP_NSR 40 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#0 Interrupt */
-					<ICU_GRP_NSR 44 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#1 Interrupt */
-					<ICU_GRP_NSR 48 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#2 Interrupt */
-					<ICU_GRP_NSR 52 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#3 Interrupt */
-					<ICU_GRP_NSR 56 IRQ_TYPE_LEVEL_HIGH>; /* RX Shared Interrupt */
-			port-id = <1>; /* pp2_port_id */
-			emac-data = <&CP110_LABEL(emac2)>;
+
+		CP110_LABEL(pcie1): pcie1@620000 {
+			compatible = "marvell,armada8k-pcie", "snps,dw-pcie";
+			reg = <CP110_PCIEx_REG0_BASE_H(1) CP110_PCIEx_REG0_BASE_L(1) 0 0x10000>,
+			      <CP110_PCIEx_REG1_BASE_H(1) CP110_PCIEx_REG1_BASE_L(1) 0 0x80000>;	/* Last 512KB of mem space */
+			reg-names = "ctrl", "config";
+			#address-cells = <3>;
+			#size-cells = <2>;
+			#interrupt-cells = <1>;
+			device_type = "pci";
+			dma-coherent;
+
+			bus-range = <0 0xff>;
+			ranges =
+				/* downstream I/O */
+				<0x81000000 0 CP110_PCIEx_IO_BASE_L(1) CP110_PCIEx_IO_BASE_H(1) CP110_PCIEx_IO_BASE_L(1) 0 0x10000
+				/* non-prefetchable memory */
+				 0x82000000 0 CP110_PCIEx_MEM_BASE_L(1) CP110_PCIEx_MEM_BASE_H(1) CP110_PCIEx_MEM_BASE_L(1) 0 0xf00000>;
+			interrupt-map-mask = <0 0 0 0>;
+			interrupt-map = <0 0 0 0 &CP110_LABEL(icu) ICU_GRP_NSR 24 IRQ_TYPE_LEVEL_HIGH>;
+			interrupts = <ICU_GRP_NSR 24 IRQ_TYPE_LEVEL_HIGH>;
+
+			num-lanes = <1>;
+			clocks = <&CP110_LABEL(syscon0) 1 11>;
 			status = "disabled";
 		};
-		CP110_LABEL(eth2): eth2@030000 {
-			interrupts = <ICU_GRP_NSR 41 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#0 Interrupt */
-					<ICU_GRP_NSR 45 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#1 Interrupt */
-					<ICU_GRP_NSR 49 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#2 Interrupt */
-					<ICU_GRP_NSR 53 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#3 Interrupt */
-					<ICU_GRP_NSR 57 IRQ_TYPE_LEVEL_HIGH>; /* RX Shared Interrupt */
-			port-id = <2>; /* pp2_port_id */
-			emac-data = <&CP110_LABEL(emac3)>;
+
+		CP110_LABEL(pcie2): pcie2@640000 {
+			compatible = "marvell,armada8k-pcie", "snps,dw-pcie";
+			reg = <CP110_PCIEx_REG0_BASE_H(2) CP110_PCIEx_REG0_BASE_L(2) 0 0x10000>,
+			      <CP110_PCIEx_REG1_BASE_H(2) CP110_PCIEx_REG1_BASE_L(2) 0 0x80000>;	/* Last 64KB of mem space */
+			reg-names = "ctrl", "config";
+			#address-cells = <3>;
+			#size-cells = <2>;
+			#interrupt-cells = <1>;
+			device_type = "pci";
+			dma-coherent;
+
+			bus-range = <0 0xff>;
+			ranges =
+				/* downstream I/O */
+				<0x81000000 0 CP110_PCIEx_IO_BASE_L(2) CP110_PCIEx_IO_BASE_H(2) CP110_PCIEx_IO_BASE_L(2) 0 0x10000
+				/* non-prefetchable memory */
+				 0x82000000 0 CP110_PCIEx_MEM_BASE_L(2) CP110_PCIEx_MEM_BASE_H(2) CP110_PCIEx_MEM_BASE_L(2) 0 0xf00000>;
+			interrupt-map-mask = <0 0 0 0>;
+			interrupt-map = <0 0 0 0 &CP110_LABEL(icu) ICU_GRP_NSR 23 IRQ_TYPE_LEVEL_HIGH>;
+			interrupts = <ICU_GRP_NSR 23 IRQ_TYPE_LEVEL_HIGH>;
+
+			num-lanes = <1>;
+			clocks = <&CP110_LABEL(syscon0) 1 12>;
 			status = "disabled";
 		};
 	};
-
-	CP110_LABEL(trng): trng@760000 {
-		compatible = "marvell,armada-8k-rng", "inside-secure,safexcel-eip76";
-		reg = <0x760000 0x7D>;
-		#address-cells = <1>;
-		#size-cells = <0>;
-		interrupts = <ICU_GRP_NSR 95 IRQ_TYPE_LEVEL_HIGH>;
-		clocks = <&CP110_LABEL(syscon0) 1 25>;
-		status = "okay";
-	};
-
-	CP110_LABEL(eip197): eip197@800000 {
-		compatible = "inside-secure,safexcel-eip197";
-		reg = <0x800000 0x200000>;
-		dma-coherent;
-		dma-bus-width = <40>;
-		interrupts = <ICU_GRP_REI 87 IRQ_TYPE_LEVEL_HIGH>,
-				<ICU_GRP_NSR 88 IRQ_TYPE_LEVEL_HIGH>,
-				<ICU_GRP_NSR 89 IRQ_TYPE_LEVEL_HIGH>,
-				<ICU_GRP_NSR 90 IRQ_TYPE_LEVEL_HIGH>,
-				<ICU_GRP_NSR 91 IRQ_TYPE_LEVEL_HIGH>,
-				<ICU_GRP_NSR 92 IRQ_TYPE_LEVEL_HIGH>;
-		interrupt-names = "eip_addr", "ring0", "ring1",
-				"ring2", "ring3", "eip_out";
-		clocks = <&CP110_LABEL(syscon0) 1 26>;
-		cell-index = <CP110_EIP197_INDEX>;
-		status = "okay";
-	};
-
-	CP110_LABEL(comphy): comphy {
-		compatible = "marvell,cp110-comphy";
-		reg = <0x441000 0x100>, <0x120000 0x6000>;
-		reg-names = "comphy", "serdes";
-		dfx-reg = <0x400280 4>;
-		#phy-cells = <2>;
-		sar-data = <&CP110_LABEL(sar)>;
-		skip_pcie_power_off;
-		status = "disabled";
-	};
-
-	CP110_LABEL(sar): sar@400200 {
-		compatible = "marvell,sample-at-reset-cp110";
-		reg = <0x400200 0x8>;
-		status = "okay";
-	};
-};
-
-CP110_LABEL(pcie_ep): pcie-ep@600000 {
-	compatible = "marvell,armada8k-pcie-ep", "snps,dw-pcie";
-	reg = <CP110_PCIE_EP_REG0_BASE_H CP110_PCIE_EP_REG0_BASE_L 0 0x4000>,
-	      <CP110_PCIE_EP_REG1_BASE_H CP110_PCIE_EP_REG1_BASE_L 0 0x80000>,
-	      <CP110_PCIE_EP_REG2_BASE_H CP110_PCIE_EP_REG2_BASE_L 0 0x1000>;
-	reg-names = "core", "lm", "shadow_core";
-	clocks = <&CP110_LABEL(syscon0) 1 13>;
-	status = "disabled";
 };
 
-CP110_LABEL(pcie0): pcie0@600000 {
-	compatible = "marvell,armada8k-pcie", "snps,dw-pcie";
-	reg = <CP110_PCIEx_REG0_BASE_H(0) CP110_PCIEx_REG0_BASE_L(0) 0 0x10000>,
-	      <CP110_PCIEx_REG1_BASE_H(0) CP110_PCIEx_REG1_BASE_L(0) 0 0x80000>;	/* Last 512KB of mem space */
-	reg-names = "ctrl", "config";
-	#address-cells = <3>;
-	#size-cells = <2>;
-	#interrupt-cells = <1>;
-	device_type = "pci";
-	dma-coherent;
-
-	bus-range = <0 0xff>;
-	ranges =
-		/* downstream I/O */
-		<0x81000000 0 CP110_PCIEx_IO_BASE_L(0) CP110_PCIEx_IO_BASE_H(0) CP110_PCIEx_IO_BASE_L(0) 0 0x10000
-		/* non-prefetchable memory */
-		 0x82000000 0 CP110_PCIEx_MEM_BASE_L(0) CP110_PCIEx_MEM_BASE_H(0) CP110_PCIEx_MEM_BASE_L(0) 0 0xf00000>;
-	interrupt-map-mask = <0 0 0 0>;
-	interrupt-map = <0 0 0 0 &CP110_LABEL(icu) ICU_GRP_NSR 22 IRQ_TYPE_LEVEL_HIGH>;
-	interrupts = <ICU_GRP_NSR 22 IRQ_TYPE_LEVEL_HIGH>;
-	num-lanes = <1>;
-	clocks = <&CP110_LABEL(syscon0) 1 13>;
-	status = "disabled";
-};
-
-CP110_LABEL(pcie1): pcie1@620000 {
-	compatible = "marvell,armada8k-pcie", "snps,dw-pcie";
-	reg = <CP110_PCIEx_REG0_BASE_H(1) CP110_PCIEx_REG0_BASE_L(1) 0 0x10000>,
-	      <CP110_PCIEx_REG1_BASE_H(1) CP110_PCIEx_REG1_BASE_L(1) 0 0x80000>;	/* Last 512KB of mem space */
-	reg-names = "ctrl", "config";
-	#address-cells = <3>;
-	#size-cells = <2>;
-	#interrupt-cells = <1>;
-	device_type = "pci";
-	dma-coherent;
-
-	bus-range = <0 0xff>;
-	ranges =
-		/* downstream I/O */
-		<0x81000000 0 CP110_PCIEx_IO_BASE_L(1) CP110_PCIEx_IO_BASE_H(1) CP110_PCIEx_IO_BASE_L(1) 0 0x10000
-		/* non-prefetchable memory */
-		 0x82000000 0 CP110_PCIEx_MEM_BASE_L(1) CP110_PCIEx_MEM_BASE_H(1) CP110_PCIEx_MEM_BASE_L(1) 0 0xf00000>;
-	interrupt-map-mask = <0 0 0 0>;
-	interrupt-map = <0 0 0 0 &CP110_LABEL(icu) ICU_GRP_NSR 24 IRQ_TYPE_LEVEL_HIGH>;
-	interrupts = <ICU_GRP_NSR 24 IRQ_TYPE_LEVEL_HIGH>;
-
-	num-lanes = <1>;
-	clocks = <&CP110_LABEL(syscon0) 1 11>;
-	status = "disabled";
-};
-
-CP110_LABEL(pcie2): pcie2@640000 {
-	compatible = "marvell,armada8k-pcie", "snps,dw-pcie";
-	reg = <CP110_PCIEx_REG0_BASE_H(2) CP110_PCIEx_REG0_BASE_L(2) 0 0x10000>,
-	      <CP110_PCIEx_REG1_BASE_H(2) CP110_PCIEx_REG1_BASE_L(2) 0 0x80000>;	/* Last 64KB of mem space */
-	reg-names = "ctrl", "config";
-	#address-cells = <3>;
-	#size-cells = <2>;
-	#interrupt-cells = <1>;
-	device_type = "pci";
-	dma-coherent;
-
-	bus-range = <0 0xff>;
-	ranges =
-		/* downstream I/O */
-		<0x81000000 0 CP110_PCIEx_IO_BASE_L(2) CP110_PCIEx_IO_BASE_H(2) CP110_PCIEx_IO_BASE_L(2) 0 0x10000
-		/* non-prefetchable memory */
-		 0x82000000 0 CP110_PCIEx_MEM_BASE_L(2) CP110_PCIEx_MEM_BASE_H(2) CP110_PCIEx_MEM_BASE_L(2) 0 0xf00000>;
-	interrupt-map-mask = <0 0 0 0>;
-	interrupt-map = <0 0 0 0 &CP110_LABEL(icu) ICU_GRP_NSR 23 IRQ_TYPE_LEVEL_HIGH>;
-	interrupts = <ICU_GRP_NSR 23 IRQ_TYPE_LEVEL_HIGH>;
-
-	num-lanes = <1>;
-	clocks = <&CP110_LABEL(syscon0) 1 12>;
-	status = "disabled";
-};
-- 
2.7.4

